airflow/contrib/kubernetes/kubernetes_request_factory/kubernetes_request_factory.py

```diff
@@ -140,6 +140,27 @@ def extract_env_and_secrets(pod, req):
                 KubernetesRequestFactory.add_secret_to_env(env, secret)
             req['spec']['containers'][0]['env'] = env
 
+        if pod.envs_from_configmaps or pod.envs_from_secrets:
+            req['spec']['containers'][0]['envFrom'] = []
+
+        for configmap in pod.envs_from_configmaps:
+            req['spec']['containers'][0]['envFrom'].append(
+                {
+                    'configMapRef': {
+                        'name': configmap
+                    }
+                }
+            )
+
+        for secret in pod.envs_from_secrets:
+            req['spec']['containers'][0]['envFrom'].append(
```

Comment:
Secrets are already supported via lines 139-140 which is adding `valueFrom: { secretKeyRef: ...` in to env`

What's the difference bewteen envFrom.valueFrom.secretKey and env.valueFrom.secretKeyRef? Do we need both?

(I'd like us not to have both - it complicates the code)

Reply:
In fact tracing this back to the KubePodOperator-

- It calls pod_generator.make_pod
- which calls `Pod()` with `envs={}` and `secrets={}`

So rather than adding a new pair of vars we should add one more (config maps) and expose envs/secrets to the `make_pod()` and Pod operator

Reply:
@ashb Reading the K8s api docs at https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#envfromsource-v1-core my understanding is that the difference is that `envFrom.valueFrom.secretKey` will populate a pod with an environment variable for all keys in a secret while `env.valueFrom.secretKeyRef` will add a single environment variable that refers to a single key in a secret.

Reply:
Ah right. Then perhaps extend the existing airflow.contrib.kubernetes.secret.Secret to make `key` optional, and if it's not set add it to `envFrom` instead?

I _think_ the API will be cleaner if we have a single `secrets` option rather than secrets (which isn't exposed to KubePodOperator but should be, and env_from_secrets.

Reply:
I think that would be cleaner as well. What do you think @galuszkak ?

Reply:
It isn't big change for me. I'm new to project so I can do whatever works for people. 

My idea initially was having 2 classes ConfigMapObject and SecretObject with names, as they were representing K8S objects, and it was pointed out, this was also leading to confusion why we need super simple classes. I see now, Secret is used in at least 2 different ways (volumes, envs etc.). My fear is that this will create some confusion for people how to use Secret. 

But I will apply this change, I believe @ashb and @pgagnon You have much better knowledge about airflow than me ;) .

Reply:
Fixed.